main.go

```diff
@@ -63,35 +67,59 @@ func main() {
 	}
 
 	for {
+
+		// Technically we should check if silenceReq has an ID here, no need to look for another
+		// active silence
+		silenceReq, err := silenceReq.FindExisting()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// If there is an existing silence ensure its not going to expiry before
+		// our healthchecks are complete
+		// TODO: Use above vars to calculate healthcheck MAX duration
+		if silenceReq.ID != "" {
+			if ok, _ := silenceReq.WillExpireBy(15 * time.Minute); ok {
+				log.Printf("Silence will expire within 15 minutes")
+				log.Printf("Creating new silence")
+				_, err = silenceReq.Build(15 * time.Minute).Send()
+				if err != nil {
+					log.Fatal(err)
+				}
+			}
+		}
+
 		if clusterTooOld(clusterBirth, maxClusterAge) {
 			log.Printf("Cluster is older than %d minutes. Exiting Cleanly.", maxClusterAge)
-			// Make sure no silence is active
-			silenceID, err := silence.FindExisting()
-			if err != nil {
-				log.Fatal(err)
-			}
-			if silenceID != "" {
-				err = silence.Remove(silenceID)
+			if silenceReq.ID != "" {
+				// TODO: There might be more than one silence to remove
+				err = silenceReq.Remove()
 				if err != nil {
 					log.Fatal(err)
 				}
 			}
 			os.Exit(0)
 		}
```

Comment:
Would it be easier to have this part before we add any silences, before the for loop starts?

Reply:
It will expire a silence if the cluster is older than "maxClusterAge" otherwise we'd continue to silence alerts forever if a pod is not ready for instance.

Reply:
Done via #5.